frontend/src/views/assignments/offer-actions.js

```diff
@@ -58,11 +59,29 @@ function OfferActionButtons(props) {
         setOfferForAssignmentRejected,
     } = props;
 
+    const [show, setShow] = React.useState(false);
+
+    const hideOfferWithdrawConfirmation = () => setShow(false);
+    const showOfferWithdrawConfirmation = () => setShow(true);
+
     function createOffers() {
         for (const assignment of selectedAssignments) {
             offerForAssignmentCreate(assignment);
         }
     }
+    function confirmOfferWithdraw() {
+        // if withdrawing multiple offers at once, show confirmation
+        if (selectedAssignments && selectedAssignments.length > 1) {
+            showOfferWithdrawConfirmation();
+        } else {
+            // does not need confirmation if only withdrawing one offer
+            executeConfirmedOfferWithdraw();
+        }
+    }
+    function executeConfirmedOfferWithdraw() {
+        withdrawOffers();
+        hideOfferWithdrawConfirmation();
```

Comment:
This can safely be called at any time, whether the dialog is open or not, so move it into `withdrawOffers` and we can eliminate a function.